CI: Flakey Arm Tests

Ref: https://github.com/pandas-dev/pandas/issues/41740

https://app.circleci.com/pipelines/github/pandas-dev/pandas/1629/workflows/fdbd00a1-21ba-45fb-8813-c4cfd2a06305/jobs/32470

________________________ TestRename.test_rename_inplace ________________________
[gw1] linux -- Python 3.7.10 /home/circleci/miniconda3/envs/pandas-dev/bin/python

self = <pandas.tests.frame.methods.test_rename.TestRename object at 0xffff47ba8510>
float_frame =                    A         B       foo         D
a3RR7rMUiG  0.656936  1.215185  0.629301  0.115558
7SCDtsxI3g -1.65....288884 -1.874402
WYIEEfxzV5  0.138755 -0.304313  0.827201 -0.329769
7WsSg25BaJ  0.632135  1.551402  1.333670 -0.128241

    def test_rename_inplace(self, float_frame):
        float_frame.rename(columns={"C": "foo"})
        assert "C" in float_frame
        assert "foo" not in float_frame
    
        c_id = id(float_frame["C"])
        float_frame = float_frame.copy()
        return_value = float_frame.rename(columns={"C": "foo"}, inplace=True)
        assert return_value is None
    
        assert "C" not in float_frame
        assert "foo" in float_frame
>       assert id(float_frame["foo"]) != c_id
E       assert 281471885249296 != 281471885249296
E        +  where 281471885249296 = id(a3RR7rMUiG    0.629301\n7SCDtsxI3g   -0.051768\nvptsj7jAlg    0.355503\nB3utB9zDIM    0.047273\nOxQGHHlFri    0.780563\nhyK...3\nw2htgi2mxz    0.198167\n3O7zwMIcBM    1.288884\nWYIEEfxzV5    0.827201\n7WsSg25BaJ    1.333670\nName: foo, dtype: float64)

pandas/tests/frame/methods/test_rename.py:191: AssertionError

--- a/pandas/tests/frame/methods/test_rename.py
+++ b/pandas/tests/frame/methods/test_rename.py
@@ -184,14 +184,16 @@ class TestRename:
         assert "C" in float_frame
         assert "foo" not in float_frame
 
-        c_id = id(float_frame["C"])
+        c_values = float_frame["C"]._values
         float_frame = float_frame.copy()
         return_value = float_frame.rename(columns={"C": "foo"}, inplace=True)
         assert return_value is None
 
         assert "C" not in float_frame
         assert "foo" in float_frame
-        assert id(float_frame["foo"]) != c_id
+        # GH 44153
+        # Used to be id(float_frame["foo"]) != c_id, but flaky in the CI
+        assert not np.shares_memory(float_frame["foo"]._values, c_values)
 
     def test_rename_bug(self):
         # GH 5344

--- a/pandas/tests/frame/methods/test_rename.py
+++ b/pandas/tests/frame/methods/test_rename.py
@@ -184,7 +184,7 @@ class TestRename:
         assert "C" in float_frame
         assert "foo" not in float_frame
 
-        c_values = float_frame["C"]._values
+        c_values = float_frame["C"]
         float_frame = float_frame.copy()
         return_value = float_frame.rename(columns={"C": "foo"}, inplace=True)
         assert return_value is None
@@ -193,7 +193,7 @@ class TestRename:
         assert "foo" in float_frame
         # GH 44153
         # Used to be id(float_frame["foo"]) != c_id, but flaky in the CI
-        assert not np.shares_memory(float_frame["foo"]._values, c_values)
+        assert float_frame["foo"] is not c_values
 
     def test_rename_bug(self):
         # GH 5344

--- a/pandas/tests/frame/methods/test_sort_index.py
+++ b/pandas/tests/frame/methods/test_sort_index.py
@@ -224,7 +224,7 @@ class TestDataFrameSortIndex:
 
         # axis=0
         unordered = frame.loc[[3, 2, 4, 1]]
-        a_values = unordered["A"]._values
+        a_values = unordered["A"]
         df = unordered.copy()
         return_value = df.sort_index(inplace=True)
         assert return_value is None
@@ -232,7 +232,7 @@ class TestDataFrameSortIndex:
         tm.assert_frame_equal(df, expected)
         # GH 44153 related
         # Used to be a_id != id(df["A"]), but flaky in the CI
-        assert not np.shares_memory(a_values, df["A"]._values)
+        assert a_values is not df["A"]._values
 
         df = unordered.copy()
         return_value = df.sort_index(ascending=False, inplace=True)

--- a/pandas/tests/frame/methods/test_sort_index.py
+++ b/pandas/tests/frame/methods/test_sort_index.py
@@ -232,7 +232,7 @@ class TestDataFrameSortIndex:
         tm.assert_frame_equal(df, expected)
         # GH 44153 related
         # Used to be a_id != id(df["A"]), but flaky in the CI
-        assert a_values is not df["A"]._values
+        assert a_values is not df["A"]
 
         df = unordered.copy()
         return_value = df.sort_index(ascending=False, inplace=True)

--- a/pandas/tests/frame/methods/test_sort_index.py
+++ b/pandas/tests/frame/methods/test_sort_index.py
@@ -224,13 +224,15 @@ class TestDataFrameSortIndex:
 
         # axis=0
         unordered = frame.loc[[3, 2, 4, 1]]
-        a_id = id(unordered["A"])
+        a_values = unordered["A"]._values
         df = unordered.copy()
         return_value = df.sort_index(inplace=True)
         assert return_value is None
         expected = frame
         tm.assert_frame_equal(df, expected)
-        assert a_id != id(df["A"])
+        # GH 44153 related
+        # Used to be a_id != id(df["A"]), but flaky in the CI
+        assert not np.shares_memory(a_values, df["A"]._values)
 
         df = unordered.copy()
         return_value = df.sort_index(ascending=False, inplace=True)

